praw/models/reddit/widgets.py

```diff
@@ -0,0 +1,533 @@
+            API_PATH['widgets'].format(subreddit=self.subreddit),
+            params={'progressive_images': progressive_images})
+
+        self._raw_items = data.pop('items')
+        super(SubredditWidgets, self).__init__(self.subreddit._reddit, data)
+
+        # reset private variables used with properties to None.
+        self._id_card = self._moderators_widget = self._sidebar = None
+        self._topbar = self._items = None
+
+        self._fetched = True
+
+    def _objectify_widget(self, data):
+        subclass = {
+            'button': ButtonWidget,
+            'calendar': Calendar,
+            'community-list': CommunityList,
+            'custom': CustomWidget,
+            'id-card': IDCard,
+            'image': ImageWidget,
+            'menu': Menu,
+            'moderators': ModeratorsWidget,
+            'subreddit-rules': RulesWidget,
+            'textarea': TextArea
+        }.get(data['kind'], Widget)
+
+        return subclass(self.subreddit, data)
+
+
+# pylint: disable=no-member
+class Widget(PRAWBase):
+    """Base class to represent a Widget."""
+
+    def __eq__(self, other):
+        """Check equality against another object."""
+        if isinstance(other, str):
+            return other.lower() == self.id.lower()
+        elif isinstance(other, Widget):
+            return self.id.lower() == other.id.lower()
+        return False
+
+    def __init__(self, subreddit, _data):
+        """Initialize the object."""
+        self.subreddit = subreddit
+        super(Widget, self).__init__(subreddit._reddit, _data)
+
+
+class ButtonWidget(Widget):
+    """Class to represent a widget containing one or more buttons.
+
+    Find an existing one:
+
+    .. code-block:: python
+
+       button_widget = None
+       widgets = reddit.subreddit('redditdev').widgets
+       for widget in widgets.sidebar:
+           if isinstance(widget, praw.models.ButtonWidget):
+               button_widget = widget
+               break
+
+       print(button_widget.buttons)
+
+    """
+
+    def __getitem__(self, item):
+        """Get a button at the specified index."""
+        return self.buttons[item]
+
+    def __init__(self, subreddit, _data):
+        """Initialize the class."""
+        buttons = _data.pop('buttons')
+        super(ButtonWidget, self).__init__(subreddit, _data)
+        self.buttons = [Button(self._reddit, button)
+                        for button in buttons]
+
+    def __len__(self):
+        """Get the number of buttons."""
+        return len(self.buttons)
+
+
+class Calendar(Widget):
+    """Class to represent a calendar widget.
+
+    Find an existing one:
+
+    .. code-block:: python
+
+       calendar = None
+       widgets = reddit.subreddit('redditdev').widgets
+       for widget in widgets.sidebar:
+           if isinstance(widget, praw.models.Calendar):
+               calendar = widget
+               break
+
+       print(calendar.googleCalendarId)
+
+    """
+
+
+# pylint: disable=no-member
```

Comment:
Does that comment work inline with the class name? Where it's at, I believe it disables that check for the remainder of the file.

Reply:
That's the behavior I've noticed. When I remove its first occurrence (above Widget), I get exactly two warnings, both about the Widget class.